osu.Game.Tests/Visual/Menus/TestSceneToolbarClock.cs

```diff
@@ -76,5 +88,37 @@ public void TestLongGameTime()
         {
             AddStep("Set game time long", () => mainContainer.Clock = new FramedOffsetClock(Clock, false) { Offset = 3600.0 * 24 * 1000 * 98 });
         }
+
+        [Test]
+        public void TestHoverBackground()
+        {
+            Box hoverBackground = null;
+
+            AddStep("Retrieve hover background", () => hoverBackground = (Box)toolbarClock.Children[0]);
+
+            AddStep("Move mouse away from clock", () => InputManager.MoveMouseTo(toolbarClock, new Vector2(0, 200)));
+            AddAssert("Hover background is not visible", () => hoverBackground.Alpha == 0);
+            AddStep("Move mouse on top of clock", () => InputManager.MoveMouseTo(toolbarClock));
+            AddAssert("Hover background is visible", () => hoverBackground.Alpha != 0);
+            AddStep("Move mouse away from clock", () => InputManager.MoveMouseTo(toolbarClock, new Vector2(0, 200)));
+            AddUntilStep("Hover background is not visible", () => hoverBackground.Alpha == 0);
+        }
+
+        [Test]
+        public void TestDisplayModeChange()
+        {
+            ToolbarClockDisplayMode initialDisplayMode = 0;
+
+            AddStep("Retrieve current state", () => initialDisplayMode = (ToolbarClockDisplayMode)clockDisplayMode.Value);
+
+            AddStep("Trigger click", () => toolbarClock.TriggerClick());
+            AddAssert("State changed from initial", () => clockDisplayMode.Value != initialDisplayMode);
+            AddStep("Trigger click", () => toolbarClock.TriggerClick());
+            AddAssert("State changed from initial", () => clockDisplayMode.Value != initialDisplayMode);
+            AddStep("Trigger click", () => toolbarClock.TriggerClick());
+            AddAssert("State changed from initial", () => clockDisplayMode.Value != initialDisplayMode);
```

Comment:
If you're wanting to add test coverage of the cycling (and enforcing the order?) then you should be testing for each enum value in order. You could start by setting the config to a known value and then checking the subsequent calls.

Reply:
> You could start by setting the config to a known value and then checking the subsequent calls.

This was my original plan, and then I realized that this actually influences what's saved for the real client's config so I got a bit worried and tried playing it safe.
I'll go change it.